js/ui/grid_core/ui.grid_core.grid_view.js

```diff
@@ -118,14 +122,46 @@ var ResizingController = modules.ViewController.inherit({
     },
 
     _toggleBestFitMode: function(isBestFit) {
-        var $element = this.component.$element();
+        var $element = this.component.$element(),
+            that = this;
+
+
+        if(this.option("advancedRendering") && this.option("columnAutoWidth")) {
+            var $rowsTable = that._rowsView._getTableElement();
 
-        $element.find("." + this.addWidgetPrefix(TABLE_CLASS)).toggleClass(this.addWidgetPrefix(TABLE_FIXED_CLASS), !isBestFit);
+            $rowsTable.css("table-layout", isBestFit ? "auto" : "fixed");
+            $rowsTable.children("colgroup").css("display", isBestFit ? "none" : "");
 
-        //B253906
-        $element.find(EDITORS_INPUT_SELECTOR).toggleClass(HIDDEN_CLASS, isBestFit);
-        $element.find(".dx-group-cell").toggleClass(HIDDEN_CLASS, isBestFit);
-        $element.find(".dx-header-row ." + this.addWidgetPrefix(TEXT_CONTENT_CLASS)).css("maxWidth", "");
+            var $headersTable = that._columnHeadersView._getTableElement();
+            var $footerTable = that._footerView._getTableElement();
+            var $headersBody;
+            if($headersTable) {
+                if(isBestFit) {
+                    $headersBody = $headersTable.children("tbody").appendTo($rowsTable);
+                } else {
+                    $headersBody = $rowsTable.children(".dx-head").appendTo($headersTable);
```

Comment:
This code needs refactoring. Code inside `if($headersTable)` and `if($footerTable)` is duplicated.